### PR TITLE
Dropdown Component

### PR DESCRIPTION
**Pull request by antictor**
- [x] Develop component code
- [x] Developer documentation
- [x] Design documentation

## Related issue
#222 

Introduction of a new Dropdown component. This component is similar to the Select except it is used to provide a dropdown of actions (a bit like a menu) rather than a form component. As such this component doesn't have a field to show the currently selected item and only response to a callback property to indicate a selected action.

### Standard dropdown
standard

### Dropdown with left icons
lefticons

### Dropdown with scrolling options
scroll

### Internet Explorer 11
window

### CODE REVIEW

**Comment by hxltrhuxze** [comment]
Can we have the label the same colour as the value state would be? so `color(neutral, 500)`?

Also, couple of notes on the Dropdown arrow:
1. When the Dropdown becomes active, can we transition the arrow to blue and rotate it 90deg?
2. Following that, can we also move the arrow `translateY(-1px)` to re-centre the icon in the middle of the bar?

EDIT: Sorry, another point. Can we make the default width of the dropdown and the sheet smaller? It's using a lot of horizontal space at the moment

**Comment by antictor** [comment]
The field fills the container width so it is up to the developer how wide they want to make it. I didn't want the developer to be limited in width.

I'll take a look at the transition.

**Comment by antictor** [comment]
spin

**Comment by hxltrhuxze** [comment]
Sorry, one more minor tweak - the dropdown arrow needs to be in its original place when the dropdown is in its default state. It should only shift up a pixel when it's active.

**Comment by antictor** [comment]
it only applies the 1px correction when it's open

**Comment by antictor** [comment]
Screenshot 2019-09-23 at 16 28 22
Screenshot 2019-09-23 at 16 28 27

File: packages/css-framework/src/components/_dropdown.scss [hunk 1]
```diff
@@ -1,124 +1,135 @@
+$borderThick: 2px;
```
Comment (m7kvqbe1) [inline packages/css-framework/src/components/_dropdown.scss]: Elsewhere the convention seems to be hyphenated variable names (`$border-thick`).

File: packages/react-component-library/src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  label: string
+}
+
+export const Dropdown: React.FC<DropdownProps> = ({
+  onSelect,
+  options,
+  label,
+}) => {
+  const onChange = (option: ValueType<DropdownOption>) => {
+    const dropdownOption = option as DropdownOption
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/Dropdown/Dropdown.tsx:21]: The as keyword in TS is new to me. So it is basically just casting the object with that type?

File: packages/react-component-library/src/components/Dropdown/DropdownLabel.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+
+import { Visibility, VisibilityOff } from '../../icons'
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/Dropdown/DropdownLabel.tsx:3]: If we can delay merging this i'll hopefully have a fix for the tree shaking issue today.

**Review by m7kvqbe1 — APPROVED** [review]
Looks good to me - might be good to hold off merging to see if we can get the tree shaking fix in pre-release.

File: packages/react-component-library/src/components/Dropdown/DropdownLabel.tsx (hunk 3 above)
Comment (antictor) [reply packages/react-component-library/src/components/Dropdown/DropdownLabel.tsx:3]: I'd rather merge it and then retrofit the improvements then it wont get held up if the other ticket does.

File: packages/react-component-library/src/components/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (antictor) [reply packages/react-component-library/src/components/Dropdown/Dropdown.tsx:21]: Yeah it just forces a cast. In this case it was needed a two types are the same but it is not possible to say that.

File: packages/css-framework/src/components/_dropdown.scss (hunk 1 above)
Comment (antictor) [reply packages/css-framework/src/components/_dropdown.scss]: Good spot

**Review by hxltrhuxze — APPROVED** [review]

**Pull request merged**